plugins/opensb_detector.py

```diff
@@ -0,0 +1,27 @@
+"""
+StarryPy OpenSB Detector Plugin
+
+Detects zstd compression for the stream and sets server configuration accordingly
+"""
+
+import asyncio
+
+from base_plugin import SimpleCommandPlugin
+from utilities import send_message, Command
+
+
+class OpenSBDetector(SimpleCommandPlugin):
+    name = "opensb_detector"
+
+    def __init__(self):
+        super().__init__()
+
+    async def activate(self):
+        await super().activate()
+
+    async def on_protocol_response(self, data, connection):
+        # self.logger.debug("Received protocol response: {} from connection {}".format(data, connection))
+        if data["parsed"]["info"]["compression"] == "Zstd":
```

Comment:
The error that occurs here when a vanilla client connects can be solved with the dict.get method. The info key will be None in this case.

Reply:
Used a .get() and check for None to avoid the exception.